### PR TITLE
Autoloader: use all of the map files to populate the global maps

### PR DESCRIPTION
**Pull request by kbrown9**
#### Changes proposed in this Pull Request:
* The latest autoloader needs to examine all of the classmap and filemap files created by the new autoloaders and populate the globals. In the `enqueue_files()` function, loop through all of the active plugins and examine the classmap and filemap files.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This changes an existing part of Jetpack.

#### Testing instructions:
* Try using multiple plugins that require the autoloader, and verify that no errors occur.

#### Proposed changelog entry for your changes:
* N/A

### CODE REVIEW

**Comment by zinigor** [comment]
It's working for me! I have tried to do the same thing I tried while testing the previous PR, and the earlier autoloader picks up the later perfectly fine. Even when I tried to instantiate an autoloaded class before the other plugin has even loaded, it worked! I don't know if this is too much black magic (should we even be allowing that, I mean), but it's definitely better than what we've had before.

File: packages/autoloader/src/functions.php [hunk 1]
```diff
@@ -134,38 +134,111 @@ function autoloader( $class_name ) {
...
+/**
+ * Returns the name of activating plugin if a plugin is activating via a request.
+ *
+ * @return String The name of the activating plugin.
+ */
+function get_activating_plugin() {
+	if ( isset( $_REQUEST['action'] ) && 'activate' === $_REQUEST['action'] ) {
+		$activating_plugin = isset( $_REQUEST['plugin'] ) ? wp_unslash( $_REQUEST['plugin'] ) : null;
+		return $activating_plugin;
+	}
```
Comment (kbrown9) [inline packages/autoloader/src/functions.php]: This probably needs a nonce verification.

Comment (zinigor) [reply packages/autoloader/src/functions.php]: You're right, but it turns out that the function that checks the validity of the nonce is not yet loaded at that time. So I have added a "check" that only requires the nonce to be present in the request, but doesn't actually check it. The request will fail later when the plugin is activated.

Comment (kbrown9) [reply packages/autoloader/src/functions.php]: This looks good to me!

**Pull request merged**